### PR TITLE
[scripts] Add old to new gwf script

### PR DESCRIPTION
**Pull request by smousy**
* [ ] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/CONTRIBUTING.md)
* [ ] Update changelog
* [ ] Update documentation

To run this script: 
``python3 old_to_new_gwf.py <path to input dir> <path to output dir>``

### CODE REVIEW

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import sys
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:1]: Please, update changelog and point this feature in readme

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                    document = parse(filegwf)
+                    updateTags(document)
+                    document.normalize()
+                    file_to_write = open(
+                        "/".join([createDir(address, to_dir), from_file]), "w+")
+                    file_to_write.write(document.toprettyxml())
+
+
+if __name__ == '__main__':
+    if len(sys.argv) == 3:
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:94]: Please, add `else` case to informate user about incorrect params

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    path_list.pop(0)
+    path = "/".join(path_list)
+    os.makedirs("/".join([to_dir, path]), 511, True)
+    return "/".join([to_dir, path])
+
+
+def translate(from_dir, to_dir):
+    if not os.path.isdir(to_dir):
+        os.mkdir(to_dir)
+    for address, dirs, files in os.walk(from_dir):
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:81]: This may throw exception if `from_dir` is not exist

Comment (NikitaZotov) [reply scripts/kb_scripts/old_to_new_gwf.py:81]: Can I pass as param no directiry path, but file path?

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        for from_file in files:
+            if from_file.endswith(".gwf"):
+                with open(os.path.join(address, from_file)) as filegwf:
+                    document = parse(filegwf)
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:85]: what if passed incorrect gwf file?

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
build after translation is failed
image

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
If I pass absolute path, the directory is created in current project directory with duplicating absolute path (лесенка)
image

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    "pair/noorient": "pair/-/-/-/noorient",
+    "pair/orient": "pair/-/-/-/orient",
+    "arc/-/-": "pair/-/-/-/orient",
+}
+
+
+def updateExistingTypesByTag(elem, tag: str):
+    elements = elem.getElementsByTagName(tag)
+    for element in elements:
+        if(tag == "arc" and element.getAttribute("type")=="node/var/perm/general"):
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:48]: Please, reformat this file

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                else:
+                    print("Incorrect file:"+from_file)
+    else:
+        print("Incorrect param input dir")
+
+if __name__ == '__main__':
+    if len(sys.argv) == 3:
+        translate(sys.argv[1], sys.argv[2])
+    else:
+        print("Incorrect params, to run this script: python3 old_to_new_gwf.py <path to input dir> <path to output dir>")
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:103]:
```suggestion
        print("Incorrect params, to run this script: python3 old_to_new_gwf.py <path to input dir> <path to output dir>")
        exit 1
```

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+                        document = parse(filegwf)
+                        updateTags(document)
+                        document.normalize()
+                        file_to_write = open(
+                            "/".join([createDir(address, to_dir), from_file]), "w+")
+                        file_to_write.write(document.toprettyxml())
+                else:
+                    print("Incorrect file:"+from_file)
+    else:
+        print("Incorrect param input dir")
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:97]:
```suggestion
        print("Incorrect param input dir")
        exit 1
```

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    for bus in buses:
+        if (bus.getAttribute("type") == ""):
+            bus.setAttribute("type", dictionary.get(bus.getAttribute("type")))
+
+
+def createDir(address, to_dir):
+    path_list = address.split("/")
+    path_list.pop(0)
+    path = "/".join(path_list)
+    os.makedirs("/".join([to_dir, path]), 511, True)
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:76]: Can you check this script for Windows. In my opinion, 511 code works for only Linux OS

File: scripts/kb_scripts/old_to_new_gwf.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+                        document.normalize()
+                        file_to_write = open(
+                            "/".join([createDir(address, to_dir), from_file]), "w+")
+                        file_to_write.write(document.toprettyxml())
+                else:
+                    print("Incorrect file:"+from_file)
+    else:
+        print("Incorrect param input dir")
+
+if __name__ == '__main__':
```
Comment (NikitaZotov) [inline scripts/kb_scripts/old_to_new_gwf.py:99]: Please, add logs to provide parsing status for users

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: scripts/kb_scripts/old_to_new_gwf.py (hunk 8 above)
Comment (smousy) [reply scripts/kb_scripts/old_to_new_gwf.py:76]: sorry, i don't have windows or virtual machine, may you do it?

**Comment by NikitaZotov** [comment]
Continued in https://github.com/ostis-ai/sc-machine/pull/333

**Pull request closed without merging**